src/main/java/org/truffleruby/core/symbol/SymbolTable.java

```diff
@@ -141,6 +145,18 @@ private DynamicObject lookupCache(Map<StringKey, SoftReference<DynamicObject>> c
         return symbolMap.values();
     }
 
+    public void rehash() {
+        // Just a cache, so we can empty and it will be repopulated as needed
+        stringToSymbolCache.clear();
+
+        symbolMap = new WeakValueCache<>(symbolMap);
```

Comment:
What's the value in doing this with a copy constructor rather than mutating the internal state?

Reply:
(same question above)